src/framed.rs

```diff
@@ -3,24 +3,53 @@ use std::{
     ops::{Deref, DerefMut},
 };
 
-use bytes::BytesMut;
-
 use crate::{
-    framed_read::FramedReadImpl, framed_write::FramedWriteImpl, fuse::Fuse, Decoder, Encoder,
-    IterSink,
+    framed_read::FramedReadImpl, framed_write::FramedWriteImpl, fuse::Fuse, Buffer, Decoder,
+    Encoder, IterSink,
 };
 
+#[cfg(feature = "futures")]
+pin_project_lite::pin_project! {
+    /// A unified `Stream` and `Sink` interface to an underlying I/O object,
+    /// using the `Encoder` and `Decoder` traits to encode and decode frames.
+    ///
+    /// # Example
```

Comment:
pin_project_lite is more clunky but doesn't bring in quote/syn. There is probably a better way to have these doc comments so they don't have to be duplicated. Might have to store them in a md file and bring them in that way